keyboards/helix/rev2/keymaps/froggy_106/keymap.c

```diff
@@ -0,0 +1,901 @@
+      ),
+  
+  /* Func
+   * ,-----------------------------------------.             ,-----------------------------------------.
+   * |RGBRST|  Hue |To101 |  RST |  Mac |  Win |             |      |      |      |      |      |      |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * | RGB1 | VAL+ |  F7  |  F8  |  F9  |To106 |             |      |      |      |      |      |      |
+   * |------+------+------+------+------+------|             |------+------+------+------+------+------|
+   * | RGB2 | VAL- |  F4  |  F5  |  F6  | F12  |             |      |      |      |      |      |      |
+   * |------+------+------+------+------+------+------+------+------+------+------+------+------+------|
+   * | RGB3 |  F10 |  F1  |  F2  |  F3  | F11  |      |      |      |      |      |      |      |      |
+   * |------+------+------+------+------+------+------+------+------+------+------+------+------+------|
+   * |RGBOFF|      |      |      |      |      |      |      |      |      |      |      |      |      |
+   * `-------------------------------------------------------------------------------------------------'
+   */
+  [_FUNC] = LAYOUT( \
+      RGBRST,RGB_HUI, TO_101,  RESET,   MAC,     WIN,                         _______,  _______,  _______,  _______,  _______,  _______, \
+      RGB1,  RGB_VAI, KC_F7,   KC_F8,   KC_F9,   TO_106,                      _______,  _______,  _______,  _______,  _______,  _______, \
+      RGB2,  RGB_VAD, KC_F4,   KC_F5,   KC_F6,   KC_F12,                      _______,  _______,  _______,  _______,  _______,  _______, \
+      RGB3,  KC_F10,  KC_F1,   KC_F2,   KC_F3,   KC_F11,   _______,  _______, _______,  _______,  _______,  _______,  _______,  _______, \
+      RGBOFF,_______, _______, _______, _______, _______,  _______,  _______, _______,  _______,  _______,  _______,  _______,  _______ \
+      )
+};
+#else
+#error "undefined keymaps"
+#endif
+
+
+#ifdef AUDIO_ENABLE
+
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+float tone_dvorak[][2]     = SONG(DVORAK_SOUND);
+float tone_colemak[][2]    = SONG(COLEMAK_SOUND);
+float tone_plover[][2]     = SONG(PLOVER_SOUND);
+float tone_plover_gb[][2]  = SONG(PLOVER_GOODBYE_SOUND);
+float music_scale[][2]     = SONG(MUSIC_SCALE_SOUND);
+#endif
+
+// define variables for reactive RGB
+//bool TOG_STATUS = false;
+int RGB_current_mode;
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+void set_mac_mode(bool enable) {
+  uint32_t old_config = user_config.raw;
+  if(enable){
+    user_config.mac_mode = true;
+    keymap_config.swap_lalt_lgui = false;
+    keymap_config.swap_ralt_rgui = false;
+  }else{
+    user_config.mac_mode = false;
+    keymap_config.swap_lalt_lgui = true;
+    keymap_config.swap_ralt_rgui = true;
+  }
+  if (user_config.raw != old_config) {
```

Comment:
You shouldn't need this. as the eeprom shouldn't re-write if the data is the same, in theory.

Reply:
Data check is removed.